zipline/test_algorithms.py

```diff
@@ -505,9 +505,10 @@ def initialize(self, asset=None, max_shares=None, max_notional=None):
 
 
 class SetDoNotOrderListAlgorithm(TradingAlgorithm):
-    def initialize(self, sid=None, restricted_list=None):
+    def initialize(self, sid=None, restricted_list=None,
+                   fail_on_violation=True):
         self.order_count = 0
-        self.set_do_not_order_list(restricted_list)
+        self.set_do_not_order_list(restricted_list, fail_on_violation)
```

Comment:
This might want to be a string/enum like `on_error` with values of `'warn'` or `'error'` rather than a bool.  Having an enum makes it easier to extend the behavior here in the future.